pennylane/hf/tapering.py

```diff
@@ -464,3 +467,106 @@ def transform_hamiltonian(h, generators, paulix_ops, paulix_sector):
     c = qml.math.stack(c)
 
     return _simplify(qml.Hamiltonian(c, o))
+
+
+def _sector_energy(sector):
+    """Computes energy of the given tapered Hamiltonian.
+
+    Args:
+        sector (pennylane.Hamiltonian): tapered Hamiltonian whose energy has to be calculated
+
+    Returns:
+        energy (float): energy of the tapered hamiltonian
+    """
+
+    if len(sector.wires) < 2:
+        energy = min(sp.linalg.eigvals(qml.utils.sparse_hamiltonian(sector).toarray()))
+    else:
+        energy = sp.sparse.linalg.eigs(qml.utils.sparse_hamiltonian(sector), k=2)[0].min()
+
+    return energy
+
+
+def find_optimal_sector(qubit_op, generators, paulix_ops, brute_force=True, num_electrons=0):
```

Comment:
I think this function should only return the correct sector and not the energy. This means that we should not perform the brute force method here since the user can always call the  `transform_hamiltonian` function with any desired sector. Also, if the user can already obtain the correct sector, then no need to do a brute force at all. Therefore, the args and return might be:

Args: generators, reference HF state 
Return: list of eigenvalues

Instead of the reference HF state, we might have `mol, core=None, active=None` or something similar.

Reply:
Agree with Soran :100: Why use an inefficient brute force method when we know a better algorithm?

Reply:
Agreed. My original intention to include brute force was for the sake of completeness. But now that `transform_hamiltonian` can already allow users to do it already, I also believe that we can skip doing it here.